### ISSUE TITLE
`change_orientation` causes sform/qform mismatch (`set_qform` seems to have no effect)

### ISSUE THREAD

**Issue by joshuacwnewton** [spinalcordtoolbox/spinalcordtoolbox#4689]
A [user on the forum](https://forum.spinalcordmri.org/t/dog-template-registration-qform-sform-error/1324/1) encountered an sform/qform mismatch error during `sct_register_to_template`, despite their input data having matching sform/qforms.

The mismatch was introduced here:

https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/ff0945d455521598474e99f31e4679f0015985b0/spinalcordtoolbox/scripts/sct_register_to_template.py#L448-L450

When I set a breakpoint before and after reorientation, I see that `im_src` has roughly the same affines (within our `check_sform` tolerance, at least) before:

```python
>>> Image(ftmp_seg).header.get_sform() - Image(ftmp_seg).header.get_qform()
array([[ 1.24708333e-08, -1.99139080e-10,  3.86632998e-11, 0.00000000e+00],
       [ 1.36003458e-10,  1.79459173e-08, -2.86510378e-10, 0.00000000e+00],
       [-5.80472270e-11,  3.40933358e-12,  1.17948034e-08, 0.00000000e+00],
       [ 0.00000000e+00,  0.00000000e+00,  0.00000000e+00, 0.00000000e+00]])
```

However, after reorientation, the difference is much larger:

```python
>>> Image(ftmp_seg).header.get_sform() - Image(ftmp_seg).header.get_qform()
array([[-1.23808742e-06, -8.25844864e-06, -1.41198199e-03, 0.00000000e+00],
       [ 8.76812599e-06, -1.79952582e-08, -4.15943024e-06, 0.00000000e+00],
       [-1.50030867e-03, -4.14610448e-06,  1.16511079e-06, 0.00000000e+00],
       [ 0.00000000e+00,  0.00000000e+00,  0.00000000e+00, 0.00000000e+00]])
```

Checking inside of the `change_orientation` function, I see that we're supposed to have a step that should eliminate any differences:

https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/96eaaaa9739859df2584868a4d07455766420a52/spinalcordtoolbox/image.py#L1165-L1166

However, when I manually run the `set_qform()` call, nothing seems to happen! There is no change in the result of `get_qform()` before and after setting it. 

I will try to find out why this is happening, and possibly report upstream to `nibabel`.

**Comment by joshuacwnewton** [comment]
OK! I have an answer.

> https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/96eaaaa9739859df2584868a4d07455766420a52/spinalcordtoolbox/image.py#L1165-L1166

Counterintuitively, these two lines will _not_ guarantee that the sform and qform will be equal. This is because the `qform` is subtly transformed into the "closest orthogonal matrix" before assignment, whereas the `sform` is assigned directly.

Meaning, if the "closest orthogonal matrix" differs significantly enough from the intended affine, then the qform and sform will also differ, and our check will fail.

To ensure identical affines (and thus prioritize the orthogonal matrix), we would need to do this instead:

```python
    im_dst.header.set_qform(im_dst_aff)
    im_dst.header.set_sform(im_dst.header.get_qform())
```

In other words, we set the `sform` to whatever the `qform` ends up as. By doing this, registration runs without error.

----

However, it would seem that @mguaypaq already thought of this when writing the `check_affines_match` check:

https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/96eaaaa9739859df2584868a4d07455766420a52/spinalcordtoolbox/image.py#L247-L252

By assigning the `sform` to the `qform` in `hdr2`, the check _should_ account for this deviation? (i.e. the sform should be made orthogonal too, and there should be no deviation)

Yet, the check still fails...
